core/src/main/java/com/oath/oak/OakUnsafeRef.java

```diff
@@ -0,0 +1,28 @@
+package com.oath.oak;
+
+import java.nio.ByteBuffer;
+import java.util.function.Function;
+
+public interface OakUnsafeRef {
```

Comment:
- This requires a chunk of external documentation. We must explain that there is an explicit tradeoff between safety and performance. 
   The user must understand that he is on his own with synchronization and bound checking. Preferably, give a couple of examples of what could go wrong. 
- Why is the name OakUnsafeRef? Shouldn't we be following the "Buffer" language everywhere?

Reply:
Regarding the second bullet, notice that this interface does not have any similarity to Java's buffers interface.
It only references to the buffer/address of the data. Hence the name.

Reply:
I am also supporting to remain with "Buffer" name! Yoav have changed OakRKeyBuffer for OakRReference and broke this rule which was a mistake. It should all be Buffers or all be References! But as currently the majority are Buffers this should be also just Unsafe Buffer. Afterwards (or already here) we can change all the naming in a consistent way. Because in "safe" variation the class is more similar to buffer, I vote for buffer-ed name. Why someone would cast Buffer to Reference? To cast Buffer to UnsafeBuffer is more reasonable....

Reply:
I agree that casting buffer to ref is nonintuitive.
But calling something a buffer when its API does not resemble a buffer in any way is confusing.
I prefer using the reference notation, but if we want to stick with Java's internal conventions, we could mimic the `DirectBuffer` naming rule and call this `OakUnsafeDirectBuffer`.
Agreed?

Reply:
OakUnsafeDirectBuffer is better than OakUnsafeRef for sure!